siriuspy/siriuspy/devices/orbit_interlock.py

```diff
@@ -1,24 +1,148 @@
-"""."""
+"""Orbit Interlock devices."""
 
 import numpy as _np
 
-from .device import Device as _Device
 from ..search import BPMSearch as _BPMSearch
+from ..util import ClassProperty as _classproperty
+from .device import Device as _Device, Devices as _Devices
 
 
-class BPMInterlock(_Device):
-    """BPM Device"""
+class BaseOrbitIntlk:
+    """Base BPM Orbit Interlock Info."""
+
+    CONV_NM2UM = 1e-3  # [nm] --> [um]
+
+    UP_2_DOWN = {
+        'M1': 'M2',
+        'C1-1': 'C1-2',
+        'C2': 'C3-1',
+        'C3-2': 'C4',
+    }
+    DOWN_2_UP = {val: key for key, val in UP_2_DOWN.items()}
+
+    __BPM_NAMES = None
+    __BPM_NICKNAMES = None
+    __BPM_POS = None
+    __BPM_DOWNSTREAM = None
+    __BPM_UPSTREAM = None
+
+    @_classproperty
+    def BPM_NAMES(cls):
+        """BPM names."""
+        if cls.__BPM_NAMES is None:
```

Comment:
why not check whether `cls.__BPM_NAMES` is not `None` and return inside the if statement. This way the large code block which is inside the `if` clause now would be at the main identation level.

Reply:
Done!